fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -992,45 +1001,39 @@ protected DateTime getCurrentDate(DateTime dateTime, TimeGrain timeGrain) {
         RequestLog.startTiming("GeneratingFilters");
         try {
             LOG.trace("Dimension Dictionary: {}", dimensionDictionary);
-            // Set of filter objects
-            Map<Dimension, Set<ApiFilter>> generated = new LinkedHashMap<>();
 
-            // Filters are optional hence check if filters are requested.
-            if (filterQuery == null || "".equals(filterQuery)) {
-                return generated;
+            if (filterQuery == null) {
+                return Collections.emptyMap();
             }
 
-            // split on '],' to get list of filters
-            List<String> apiFilters = Arrays.asList(filterQuery.split(COMMA_AFTER_BRACKET_PATTERN));
-            for (String apiFilter : apiFilters) {
-                ApiFilter newFilter;
-                try {
-                    newFilter = new ApiFilter(apiFilter, table, dimensionDictionary);
-                } catch (BadFilterException filterException) {
-                    throw new BadApiRequestException(filterException.getMessage(), filterException);
-                }
+            ApiFiltersListListener apiFiltersExtractor = new ApiFiltersListListener(table, dimensionDictionary);
 
-                if (!BardFeatureFlag.DATA_FILTER_SUBSTRING_OPERATIONS.isOn()) {
-                    FilterOperation filterOperation = newFilter.getOperation();
-                    if (filterOperation.equals(FilterOperation.startswith)
-                            || filterOperation.equals(FilterOperation.contains)
-                            ) {
-                        throw new BadApiRequestException(
-                                ErrorMessageFormat.FILTER_SUBSTRING_OPERATIONS_DISABLED.format()
-                        );
+            ANTLRInputStream input = new ANTLRInputStream(filterQuery);
+            FiltersLex lexer = new FiltersLex(input);
+            lexer.removeErrorListeners();
+            lexer.addErrorListener(ExceptionErrorListener.INSTANCE);
+            CommonTokenStream tokens = new CommonTokenStream(lexer);
+            FiltersParser parser = new FiltersParser(tokens);
+            parser.removeErrorListeners();
+            parser.addErrorListener(ExceptionErrorListener.INSTANCE);
 
-                    }
-                }
-                Dimension dim = newFilter.getDimension();
-                if (!generated.containsKey(dim)) {
-                    generated.put(dim, new LinkedHashSet<>());
+            try {
+                try {
+                    FiltersParser.FiltersContext tree = parser.filters();
```

Comment:
[ditto](https://github.com/yahoo/fili/pull/144/files#r98962547).

Also, it looks like this is pretty much identical to the linked code in the `ApiFilter`. We may want to pull this logic out into a utility method. Ideally, it'd be something that could take a `ParseTreeListener`, and an AntlrInputStream and that populates the listener with the appropriate values. That should make it easier to integrate other ANTRL grammars as well.